src/FffGcodeWriter.cpp

```diff
@@ -362,6 +362,17 @@ void FffGcodeWriter::setInfillAndSkinAngles(SliceMeshStorage& mesh)
         }
     }
 
+    if (mesh.roofing_angles.size() == 0)
+    {
+        mesh.roofing_angles = mesh.getSettingAsIntegerList("roofing_angles");
+        if (mesh.roofing_angles.size() == 0)
+        {
+            // user has not specified any infill angles so use defaults
+            mesh.roofing_angles.push_back(45);
+            mesh.roofing_angles.push_back(135);
+        }
+    }
```

Comment:
I don't want to make Roofing a child setting of anything, because then we have to introduce a parent setting for each place we introduced a new Roofing setting.